File: src/frontend/src/app/components/financeiro/centro-custo/model/centro-custo-dto.ts
export class CentroCustoDTO {
  id?: string;
  nome: string;
  centroResultado?: boolean;
  unidadeNegocioId: string;
  createdAt?: string;
  updatedAt?: string;
  createdBy?: string;
  updatedBy?: string;

  constructor(
    id: string | undefined,
    nome: string,
    centroResultado: boolean | undefined,
    unidadeNegocioId: string,
    createdAt: string | undefined,
    updatedAt: string | undefined,
    createdBy: string | undefined,
    updatedBy: string | undefined
  ) {
    this.id = id;
    this.nome = nome;
    this.centroResultado = centroResultado;
    this.unidadeNegocioId = unidadeNegocioId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/centrocusto/CentroCustoDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.centrocusto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class CentroCustoDTO implements DTO {
    private UUID id;
    private String nome;
    private Boolean centroResultado;
    private UUID unidadeNegocioId;
    private String unidadeNegocioNome;
    private String unidadeNegocioCodigo;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
